### PR TITLE
OLED-246 - refactor and add eks enricher for account id and account group

### PR DESCRIPTION
**Pull request by jaskerv**
*Issue #, if available:*
https://canvadev.atlassian.net/browse/OLED-246
*Description of changes:*
Refactors how enricher works.
- Separate types of enricher based off deployment type (ecs, eks)
- Adds EKS enricher for account ids and account group

### CODE REVIEW

File: enricher/mappings/mappings.go [hunk 1]
```diff
@@ -0,0 +1,11 @@
+package mappings
+
+const (
+	RESOURCE_CLOUD_ACCOUNT_ID = "cloud.account.id"
+	RESOURCE_ACCOUNT_GROUP    = "account_group"
```
Comment (jaskerv) [inline enricher/mappings/mappings.go]: 👋 @wimnat 

How should we label the account group? Would this be fine under `resource`

**Review by foadnh — APPROVED** [review]
LGTM.
I also created a similar PR with different tags: https://github.com/Canva/amazon-kinesis-streams-for-fluent-bit/pull/9

File: enricher/mappings/mappings.go (hunk 1 above)
Comment (wimnat) [reply enricher/mappings/mappings.go]: I think it works under `resource` alongside the account ID. 👍

Comment (jaskerv) [reply enricher/mappings/mappings.go]: We decided that the account group function will fall under as `canva.account.function`

**Pull request merged**